src/login/pam_systemd.c

```diff
@@ -366,6 +384,11 @@ _public_ PAM_EXTERN int pam_sm_open_session(
 
         remote = !isempty(remote_host) && !is_localhost(remote_host);
 
+        memory_max = pam_getenv(handle, "SESSION_LIMIT_MEMORY_MAX");
+        tasks_max = pam_getenv(handle, "SESSION_LIMIT_TASKS_MAX");
+        cpu_weight = pam_getenv(handle, "SESSION_LIMIT_CPU_WEIGHT");
+        io_weight = pam_getenv(handle, "SESSION_LIMIT_IO_WEIGHT");
```

Comment:
Hmm, so here's the thing, using env vars will work, but I think we can do better. Setting env vars here means they'll also be set for the actual session processes, which is not particularly desirable I guess. Now we could clean up all variables we use before proceeding to the next PAM module, and that'd work fine, but I think there's an even better approach:

https://linux.die.net/man/3/pam_set_data

We already use that for passing data between our session start and end hooks, but this concept works also between various modules. Hence, I'd propose that we use:

`pam_get_data(handle, "systemd.memory_max", &memory_max)` 

and so on. The data fields set that way won't end up in the env block, they are entirely internal to the PAM context.

While pam_get_data() in theory allows passing binary data I'd probably stick to strings only for this, after all this is going to be an API between our PAM module and others, and hence sticking to a very generic, non-binary data type is the better approach.

Reply:
PAM modules earlier in the stack can, which I though this was about, no?

What specific usecase do you have in mind?

Reply:
That's right.

Reply:
but in that case external processes/users wouldn't set this anyway, but only other PAM modules (such as the sssd ones) prior in the same PAM stack